app/uk/gov/hmrc/essttpstubs/repo/PegaTokenRepo.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.essttpstubs.repo
+
+import com.google.inject.{Inject, Singleton}
+import org.mongodb.scala.model.{IndexModel, IndexOptions, Indexes}
+import org.mongodb.scala.result.InsertOneResult
+import uk.gov.hmrc.essttpstubs.model.PegaOauthToken
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+@SuppressWarnings(Array("org.wartremover.warts.Any"))
+@Singleton
+final class PegaTokenRepo @Inject() (
+    mongoComponent: MongoComponent
+)(implicit ec: ExecutionContext)
+  extends PlayMongoRepository[PegaOauthToken](
+    mongoComponent = mongoComponent,
+    collectionName = "essttp-stubs-pega-token",
+    domainFormat   = PegaOauthToken.pegaFormat,
+    indexes        = PegaTokenRepo.indexes(30.minutes.toSeconds),
```

Comment:
we should probably base this off the value of `pegaTokenExpiryTime` in `appConfig`. If you inject `appConfig` then you should have access to it here. If we configure the stub to have an expiry time of more than 30 mins then funny things will probably start happening. We could set it to something like `pegaTokenExpiryTime * 2` or something to make sure it lives long enough